exporter/azuredataexplorerexporter/adx_exporter.go

```diff
@@ -50,32 +56,46 @@ func (e *adxMetricsProducer) metricsDataPusher(ctx context.Context, metrics pmet
 		if err != nil {
 			e.logger.Error("Error performing serialization of data.", zap.Error(err))
 		}
-		ingestreader := bytes.NewReader(adxmetricjsonbytes)
-		if e.managedingest != nil {
-			if _, err := e.managedingest.FromReader(context.Background(), ingestreader, e.ingestoptions...); err != nil {
-				e.logger.Error("Error performing managed data ingestion.", zap.Error(err))
-				return err
-			}
-		} else {
-			if _, err := e.queuedingest.FromReader(context.Background(), ingestreader, e.ingestoptions...); err != nil {
-				e.logger.Error("Error performing queued data ingestion.", zap.Error(err))
-				return err
-			}
+		adxmetricjsonbytes = append(adxmetricjsonbytes, nextline...)
+		metricsByteArray = append(metricsByteArray, adxmetricjsonbytes...)
+	}
+	if len(metricsByteArray) != 0 {
+		if err := e.ingestData(metricsByteArray); err != nil {
+			return err
 		}
+		metricsByteArray = metricsByteArray[:0]
 	}
 	metricsflushed := len(transformedadxmetrics)
 	e.logger.Sugar().Infof("Flushing %d metrics to sink", metricsflushed)
 	return nil
 }
 
-func (amp *adxMetricsProducer) Close(context.Context) error {
+func (e *adxDataProducer) ingestData(b []byte) error {
+
+	ingestreader := bytes.NewReader(b)
+	if e.managedingest != nil {
+		if _, err := e.managedingest.FromReader(context.Background(), ingestreader, e.ingestoptions...); err != nil {
+			e.logger.Error("Error performing managed data ingestion.", zap.Error(err))
+			return err
+		}
+	} else {
+		if _, err := e.queuedingest.FromReader(context.Background(), ingestreader, e.ingestoptions...); err != nil {
+			e.logger.Error("Error performing queued data ingestion.", zap.Error(err))
+			return err
+		}
+	}
+	return nil
+}
+
+func (amp *adxDataProducer) Close(context.Context) error {
```

Comment:
Needs to change a bit and do a bit more. If the managedIngest is not nil that gets closed , else queuedIngest is closed. Since the client is no longer shared , close the client as well